### PR TITLE
restrict access to management functions in strategies

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
- ♻️ extract IAccessControlViewer interface from IProject
- ✨ create an access_control_viewer module that forwards calls to the project
- ♻️ use access_control_viewer library in access_control strategy
- ⚰️ remove dead code
- ✨ initialize access control in contribution contract
- ♻️ add useful functions in access control viewer
- ✨ restrict access to change_gate function
- ✨ restrict access to close and reopen functions

### CODE REVIEW

File: contracts/onlydust/marketplace/interfaces/access_control.cairo [hunk 1]
```diff
@@ -0,0 +1,9 @@
+%lang starknet
+
+@contract_interface
+namespace IAccessControlViewer {
+    func is_lead_contributor(contributor_account_address) -> (res: felt) {
```
Comment (Bernardstanislas) [inline contracts/onlydust/marketplace/interfaces/access_control.cairo:5]: `is_project_lead`?

**Review by Bernardstanislas — APPROVED** [review]
I am not sure the interface is consistent with the implementation in terms of function naming

File: contracts/onlydust/marketplace/interfaces/access_control.cairo (hunk 1 above)
Comment (AnthonyBuisset) [reply contracts/onlydust/marketplace/interfaces/access_control.cairo:5]: yes, but impact is greater, so I kept this in my TODO for another PR

**Comment by AnthonyBuisset** [comment]
> I am not sure the interface is consistent with the implementation in terms of function naming

it is, but agreed it is not very clear.
`IAccessControlViewer` is the interface implemented by the project
the `AccessControlViewer` namespace is a helper that forwards calls to the project, that implements `IAccessControlViewer`. It behaves as a proxy

I will do a dedicated PR to rename all functions so that we have a real proxy

**Pull request merged**